### PR TITLE
`amp-toolbox-cache-url`: Added Rollup for Browser vs. Server module builds

### PR DESCRIPTION
**Pull request by torch2424**
relates to #100 

This adds rollup as a build system for clean browser vs server builds for `amp-toolbox-cache-url`.

Most changes come from updates in the `package-lock.json`. 

Would appreciate a review from @erwinmombay on this 😄

### CODE REVIEW

**Comment by torch2424** [comment]
Made a few extra commits, but this is now good to go @sebastianbenz and @erwinmombay 👍

**Comment by torch2424** [comment]
Travis is failing because of node v6 support. Since the latest LTS is version 8, how would everyone feel about making this support node LTS? 

This is so that we can use `npx` to have our build system part of the project, as opposed to globally installed. Though i can remove this if anyone has any reservations.

File: cache-url/package.json [hunk 1]
```diff
@@ -23,13 +26,19 @@
...
   "dependencies": {
     "mime-types": "^2.1.19",
     "punycode": "^2.1.1"
+  },
+  "devDependencies": {
+    "@ampproject/rollup-plugin-closure-compiler": "^0.7.3",
+    "eslint": "5.6.1",
+    "eslint-config-google": "0.10.0",
+    "jasmine": "3.2.0",
+    "rollup": "^0.66.2",
```
Comment (erwinmombay) [inline cache-url/package.json]: lets stay consistent and not use `^` ranges

Comment (torch2424) [reply cache-url/package.json]: Done, thanks for the review! 😄

File: cache-url/package.json [hunk 2]
```diff
@@ -23,13 +26,19 @@
     "url": "https://github.com/ampproject/amp-toolbox/issues"
   },
   "homepage": "https://github.com/ampproject/amp-toolbox#readme",
-  "devDependencies": {
-    "eslint": "5.6.1",
-    "eslint-config-google": "0.10.0",
-    "jasmine": "3.2.0"
-  },
   "dependencies": {
     "mime-types": "^2.1.19",
```
Comment (sebastianbenz) [inline cache-url/package.json]: I think we can remove this dependency now

File: cache-url/rollup.config.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
+import resolve from 'rollup-plugin-node-resolve';
```
Comment (sebastianbenz) [inline cache-url/rollup.config.js]: Please add a license header

File: cache-url/package.json (hunk 2 above)
Comment (torch2424) [reply cache-url/package.json]: Done with the master merge 👍

File: cache-url/rollup.config.js (hunk 3 above)
Comment (torch2424) [reply cache-url/rollup.config.js]: Done.

**Comment by sebastianbenz** [comment]
@torch2424 I thought npx is available since Node 5.x. We might have to install it explicitly.

**Comment by sebastianbenz** [comment]
Installing npx globally in travis should be fine. Jasmine and eslint are installed locally.

**Comment by torch2424** [comment]
@sebastianbenz Every thing after having `npx` installed 😄 . Don't have merge access so whenever you get the chance. 

I'll branch off of this branch and start the final clean up PR with in browser testing.

**Comment by sebastianbenz** [comment]
Thanks a lot Aaron!

**Pull request merged**